src/user-event/utils/dispatch-event.ts

```diff
@@ -15,6 +17,30 @@ export function dispatchHostEvent(
   element: ReactTestInstance,
   eventName: string,
   event: unknown
+) {
+  const handler = findEnabledEventHandler(element, eventName);
+  if (!handler) {
+    return;
+  }
+
+  act(() => {
+    handler(event);
+    flushMicroTasks();
```

Comment:
We don't call `flushMicroTasks` in `fireEvent` if I'm correct, should it be called there as well? Is it needed at all if we're using act? If so could you provide a test case that would fail if we didn't call this?

Reply:
I've spotted error, as flushing microtasks requires `await`. Thanks for prompting me. 

It is not feasible to flush micro tasks in `fireEvent` as it would require signature change to `async`. BTW sync `act` does not have ability to flush micro tasks.  Beside that `fireEvent` it emits only a single event, and user has an option to flush the tasks manually if needed.

Reply:
I've done a background research on RTL's UE approach. So they use [`dispatchEvent`](https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/dispatchEvent) which actually delivers event synchronously in contrast to "native" browser events which are using event loop. 

So if sync events are good enough for UE, then we should not bother too much with going async here.